src/aragonjs-wrapper.js

```diff
@@ -72,24 +210,124 @@ const initWrapper = async (
     throw err
   }
 
-  const { apps, forwarders, transactions } = wrapper
+  const subscriptions = subscribe(
+    wrapper,
+    {
+      onApps,
+      onForwarders,
+      onTransaction,
+    },
+    {
+      ipfsConf,
+    }
+  )
 
-  const subscriptions = {
-    apps: apps.subscribe(apps => {
-      const frontendApps = prepareFrontendApps(apps)
-      onApps(frontendApps, apps)
-    }),
-    forwarders: forwarders.subscribe(onForwarders),
-    transactions: transactions.subscribe(onTransaction),
+  wrapper.connectAppIFrame = (iframeElt, proxyAddress) => {
+    const provider = new providers.WindowMessage(iframeElt.contentWindow)
+    const result = wrapper.runApp(provider, proxyAddress)
+    if (subscriptions.connectedApp) {
+      subscriptions.connectedApp.unsubscribe()
+    }
+    subscriptions.connectedApp = result.shutdown
+    return result
   }
 
   wrapper.cancel = () => {
     Object.values(subscriptions).forEach(subscription => {
-      subscription.unsubscribe()
+      if (subscription) {
+        subscription.unsubscribe()
+      }
     })
   }
 
   return wrapper
 }
 
+const templateParamFilters = {
+  democracy: (
+    // supportNeeded: Number between 0 (0%) and 1 (100%).
+    // minAcceptanceQuorum: Number between 0 (0%) and 1 (100%).
+    // voteDuration: Duration in seconds.
+    { supportNeeded, minAcceptanceQuorum, voteDuration },
+    account
+  ) => {
+    const tokenBase = Math.pow(10, 18)
+    const percentageBase = Math.pow(10, 18)
+    const holders = [{ address: account, balance: 1 }]
+
+    const [accounts, stakes] = holders.reduce(
```

Comment:
This bit seems unnecessary, since `holders` holds just one element. But I thought the democracy template allowed more than one holder; is that a future<sup>TM</sup> thing?